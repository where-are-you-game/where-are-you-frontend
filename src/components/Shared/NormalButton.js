import React from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

import { buttonStyle } from "../../styles";

const Button = styled.button`
  ${buttonStyle}
`;

function NormalButton({ title, color, handleClick }) {
  return (
    <Button
      type="button"
      color={color}
      onClick={handleClick}
    >
      {title}
    </Button>
  );
}

NormalButton.propTypes = {
  title: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default NormalButton;
