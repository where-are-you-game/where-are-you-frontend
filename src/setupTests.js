import React from "react";

import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { createStore } from "redux";
import { ThemeProvider } from "styled-components";

import { ModalProvider } from "./contexts/ModalContext";
import reducer from "./reducers";
import GlobalStyles from "./styles";
import GlobalFonts from "./styles/fonts";
import theme from "./styles/theme";

export const store = createStore(reducer);

export const history = createMemoryHistory({
  initialEntries: ["/"]
});

export const customRender = (children) => (
  render((
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <GlobalStyles />
          <GlobalFonts />
          <ModalProvider>
            {children}
          </ModalProvider>
        </ThemeProvider>
      </Provider>
    </Router>
  ))
);

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element) => element);
});
