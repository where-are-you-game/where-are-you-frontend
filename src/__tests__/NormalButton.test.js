import React from "react";

import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import NormalButton from "../components/Shared/NormalButton";
import { customRender } from "../setupTests";

describe("NormalButton", () => {
  it("matches snapshot", () => {
    const { container } = customRender(
      <NormalButton title="버튼" color="black" handleClick={() => "click"} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should fire event", () => {
    jest.spyOn(global.console, "log");
    customRender(
      <NormalButton title="button" color="black" handleClick={() => console.log("click")} />
    );

    const button = screen.getByRole("button", { name: "button" });

    userEvent.click(button);

    expect(console.log).toHaveBeenCalledWith("click");
  });
});
