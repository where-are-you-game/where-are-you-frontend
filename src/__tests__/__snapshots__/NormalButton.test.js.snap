// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NormalButton matches snapshot 1`] = `
<div>
  <button
    class="sc-hKFxyN cEASwB"
    color="black"
    name="버튼"
    type="button"
  >
    <span>
      버튼
    </span>
  </button>
</div>
`;
